cmake/ExperimentalEmscripten.cmake

```diff
@@ -0,0 +1,112 @@
+#   - Exception catching is disabled, this exception cannot be caught.
+#
+# `-sSUPPORT_LONGJMP=wasm`
+# Enables experimental support for LONGJMP in functions which may throw exceptions.
+# Without this, boost would not compile (LLVM errored out). 
+#
+# `-sEXIT_RUNTIME=1`
+# Indicates the runtime environment (node) should exit when our main() returns.
+#
+# `-sINITIAL_MEMORY=300MB` 
+# Without this, the tests OOM. The first sign is an error out with the following:
+#
+#   Pthread 0x0058faf8 sent an error! http://localhost:6931/<test>: RuntimeError: unreachable executed
+#
+# If we run the problematic test in a browser with `emrun`, we can see some javascript errors that explain:
+#
+#   Aborted(Cannot enlarge memory arrays to size 17457152 bytes (OOM). Either
+#   (1) compile with -sINITIAL_MEMORY=X with X higher than the current value 16777216, 
+#   (2) compile with -sALLOW_MEMORY_GROWTH which allows increasing the size at runtime, or 
+#   (3) if you want malloc to return  NULL (0) instead of this abort, compile with -sABORTING_MALLOC=0)
+#
+# Note that (2) is not an option because pthread cannot yet be combined with -sALLOW_MEMORY_GROWTH:
+# See https://github.com/WebAssembly/design/issues/1271
+# Smaller values (150MB, 200MB) produce intermittent failures. 300MB was chosen to give enough headroom for
+# tests written in the future. 
+#
+# `-sPTHREAD_POOL_SIZE=32` 
+# Without this, the tests deadlock. Lower values were tested. 
+# 8 threads deadlocked consistently, 16 threads passed consistently. 
+# 32 was chosen to give enough headroom for tests written in the future. 
+#
+# `-sPROXY_TO_PTHREAD`
+# This flag wraps our executable's main function in a pthread. 
+# Without this, we exhaust the thread pool very quickly. The error looks like this:
+#
+#   Tried to spawn a new thread, but the thread pool is exhausted.
+#   This might result in a deadlock unless some threads eventually exit or the code explicitly breaks out to the event loop.
+# 
+# You can read more about the setting here: https://emscripten.org/docs/porting/pthreads.html#blocking-on-the-main-browser-thread
+#
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread -fwasm-exceptions -sSUPPORT_LONGJMP=wasm")
+set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -pthread -fwasm-exceptions -sEXIT_RUNTIME=1 -sINITIAL_MEMORY=300MB -sPTHREAD_POOL_SIZE=32 -sPROXY_TO_PTHREAD")
+set(STLAB_WASM_NODE_JS_MIN_VERSION "16.16.0")
+
+#
+# Note: https://www.npmjs.com/package/wasm-check is a useful utility to find 
+# which --experimental-wasm-xxx flags are supported by node.
+#
+set(STLAB_REQUIRED_WASM_EXPERIMENTAL_FLAGS "--experimental-wasm-threads;--experimental-wasm-eh")
+
+#
+# Check if NODE_JS_EXECUTABLE (found by find_program() in Emscripten.cmake) is recent enough for STLab.
+# Set CMAKE_CROSSCOMPILING_EMULATOR to a sufficiently recent node + required experimental flags.
+#
+if (NODE_JS_EXECUTABLE)
```

Comment:
Similar suggestion to the above (and FFTI) in that `if( NOT NODE_JS_EXECUTABLE )` allows for less indentation and error messages closer to the precondition check.